src/main/mw.c

```diff
@@ -246,24 +247,20 @@ void annexCode(void)
         rcCommand[PITCH] = rcCommand_PITCH;
     }
 
-    if (feature(FEATURE_VBAT | FEATURE_CURRENT_METER)) {
-        vbatCycleTime += cycleTime;
-        if (!(++vbatTimer % VBATFREQ)) {
-
-            if (feature(FEATURE_VBAT)) {
-                updateBatteryVoltage();
-                batteryState = calculateBatteryState();
-                //handle beepers for battery levels
-                if (batteryState == BATTERY_CRITICAL)
-                    beeper(BEEPER_BAT_CRIT_LOW);    //critically low battery
-                else if (batteryState == BATTERY_WARNING)
-                    beeper(BEEPER_BAT_LOW);         //low battery
-            }
+    if (feature(FEATURE_VBAT)) {
+        /* currentTime will rollover @ 70 minutes */
+        if ((currentTime - vbatLastServiced) >= VBATINTERVAL) {
```

Comment:
I'm going to go with this instead

static uint32_t vbatNextService = 0;
...
    if (feature(FEATURE_VBAT)) {
        if (currentTime >= vbatNextService){
           vbatNextService = currentTime + VBATINTERVAL;
            updateBattery();
        }
    }


It's a lot cleaner IMHO